web/src/pages/loading.tsx

```diff
@@ -29,6 +29,19 @@ const LoadingPage = (): ReactElement => {
       return;
     }
 
+    if (typeof window !== "undefined") {
+      const returnToLink = sessionStorage.getItem("returnToLink");
+      if (returnToLink) {
+        sessionStorage.removeItem("returnToLink");
+        if (Object.values(ROUTES).includes(returnToLink)) {
+          router.push(returnToLink);
+        } else {
+          router.push(ROUTES.dashboard);
+        }
+        return;
+      }
+    }
+
```

Comment:
take a look at the useMountEffectWhenDefined code below, there is an else statement that routes the user to the dashboard, I don't think we need it here.